front/lib/api/data_sources.ts

```diff
@@ -89,12 +90,7 @@ export async function hardDeleteDataSource(
   auth: Authenticator,
   dataSource: DataSourceResource
 ) {
-  if (!auth.isBuilder()) {
-    return new Err({
-      code: "unauthorized_deletion",
-      message: "Only builders can destroy data sources.",
-    });
-  }
+  assert(auth.isBuilder(), "Only builders can delete data sources.");
```

Comment:
are we ok to replace a Result by a throw here ? why not in `softDeleteDataSourceAndLaunchScrubWorkflow` then ?

Reply:
👀

Reply:
So `softDeleteXXXX` can be called from the API(s) hence the need to have proper handling, `hardDeleteXXX` on the other side, should only be called from scrubbing workflows.